CMakeLists.txt

```diff
@@ -94,6 +94,16 @@ cmake_dependent_option(IREE_ENABLE_COMPILATION_BENCHMARKS "Builds IREE benchmark
 
 option(IREE_BUILD_MICROBENCHMARKS "Builds IREE microbenchmark suites." OFF)
 
+# For development, builds LLVM (and in the future) the whole compiler as
+# individual shared libraries similar to if passing -DBUILD_SHARED_LIBS=ON
+# to a standalone LLVM build. This can dramatically reduce linking time and
+# makes the management of some dependencies more strict.
+option(IREE_COMPILER_BUILD_SHARED_LIBS "Enables BUILD_SHARED_LIBS CMake mode for LLVM and the compiler (this is only suitable for development)" OFF)
+
+# Must be defined as an option (CMake does not do it automatically), even though
+# we override it for different parts of the tree.
+option(BUILD_SHARED_LIBS "Instructs CMake to build libraries as shared if possible" OFF)
```

Comment:
I think this would perhaps be more appropriate as a custom property? https://cmake.org/cmake/help/v3.8/command/define_property.html

Reply:
This is actually a CMake thing and believe it or not, this is how it is done. I suspect that future them would have agreed with you at the time it was done.

https://cmake.org/cmake/help/latest/variable/BUILD_SHARED_LIBS.html